plugins/woocommerce/includes/class-wc-tracker.php

```diff
@@ -381,10 +410,9 @@ public static function get_product_counts() {
 	 * @return array
 	 */
 	private static function get_order_counts() {
-		$order_count      = array();
-		$order_count_data = wp_count_posts( 'shop_order' );
+		$order_count = array();
 		foreach ( wc_get_order_statuses() as $status_slug => $status_name ) {
-			$order_count[ $status_slug ] = $order_count_data->{ $status_slug };
+			$order_count[ $status_slug ] = wc_orders_count( $status_slug );
```

Comment:
Would this count only those orders with the `shop_order` type for the status? What happens in the case of `shop_refund`? I am not sure if any refunded orders would show any difference in data.
These are my initial thoughts.
I shall check that soon.

Reply:
Good catch, currently, it excludes shop order refunds currently, [as they are marked to not be counted.](https://github.com/woocommerce/woocommerce/blob/953a8f5c30c9041cf5d9a2eeac0ab7385a91b16d/plugins/woocommerce/includes/class-wc-post-types.php#L440) However, in the earlier code, we are not counting them either, is this expected?

Another notable difference between order counting is that, earlier we were only counting posts with `shop_order` type, but now we will also count order types that are registered with the count param set to true.

Reply:
Thanks for the pointer. Yes, we're not counting in the earlier code. If they're not counted, it should be fine. I did one test with a refunded order and found that in both the cases, the count was the same.